jib-core/src/test/java/com/google/cloud/tools/jib/tar/TarExtractorTest.java

```diff
@@ -79,4 +89,64 @@ public void testExtract_symlinks() throws URISyntaxException, IOException {
     Assert.assertTrue(Files.isSymbolicLink(destination.resolve("directory-symlink")));
     Assert.assertTrue(Files.isSymbolicLink(destination.resolve("directory1/file-symlink")));
   }
+
+  @Test
+  public void testExtract_modificationTimePreserved() throws URISyntaxException, IOException {
+    Path source = Paths.get(Resources.getResource("core/extract.tar").toURI());
+    Path destination = temporaryFolder.getRoot().toPath();
+
+    TarExtractor.extract(source, destination);
+
+    assertThat(Files.getLastModifiedTime(destination.resolve("file A")))
+        .isEqualTo(FileTime.from(Instant.parse("2019-08-01T16:13:09Z")));
+    assertThat(Files.getLastModifiedTime(destination.resolve("file B")))
+        .isEqualTo(FileTime.from(Instant.parse("2019-08-01T16:12:00Z")));
+    assertThat(
+            Files.getLastModifiedTime(
+                destination.resolve("folder").resolve("nested folder").resolve("file C")))
+        .isEqualTo(FileTime.from(Instant.parse("2019-08-01T16:12:21Z")));
+  }
+
+  @Test
+  public void testExtract_symlinks_modificationTimePreserved()
+      throws URISyntaxException, IOException {
+    Path source = Paths.get(Resources.getResource("core/symlinks.tar").toURI());
+    Path destination = temporaryFolder.getRoot().toPath();
+
+    Map<String, FileTime> originalModificationTimeMap = createModificationTimeMap(source);
+
+    TarExtractor.extract(source, destination);
+
+    // Validate that the symlink's modification time is set to the target's.
```

Comment:
This seems weird. The timestamp of a symlink should be different from that of its target.